src/Core/Hook/HookDispatcher.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Hook;
+
+use PrestaShopBundle\Service\Hook\HookDispatcher as HookDispatcherService;
+
+/**
+ * Class HookDispatcher is responsible for dispatching hooks
+ */
+final class HookDispatcher implements HookDispatcherInterface
+{
+    /**
+     * @var HookDispatcherService
+     */
+    private $hookDispatcherService;
+
+    /**
+     * @param HookDispatcherService $deprecatedHookDispatcherService
+     */
+    public function __construct(HookDispatcherService $deprecatedHookDispatcherService)
+    {
+        $this->hookDispatcherService = $deprecatedHookDispatcherService;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function dispatch(HookInterface $hook)
+    {
+        $this->hookDispatcherService->dispatchForParameters(
+            $hook->getName(),
+            $hook->getParameters()
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function dispatchWithParameters($hookName, array $hookParameters = [])
+    {
+        $this->dispatch(new Hook($hookName, $hookParameters));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function dispatchRendering(HookInterface $hook)
```

Comment:
Should be called `->dispatch` too. But check for class interface or capability to use render or not.
If we change the hook capability, we will need to change the method name everywhere, when the first aim is to dispatch a hook. No?
Plus, if you stay with `dispatchRendering` method, the parameter isn't the right one, must be `RenderedHookInterface`, same for parent interface and doc block :)

Reply:
@PierreRambaud thanks for your comment! :)

However, Im not sure if I get it correctly. Why do you think parameter is wrong for `dispatchRendering`? I think that both types of hooks have same structure (name and parameters) despite whether it's a display or an action hook.

The main difference between `dispatch()` and `dispatchRendering()` is that you dont expect anything to be returned from `dispatch()` (right?) but you always know that `dispatchRendering` will return `RenderedHookInterface`.

Reply:
I just think like for Symfony and `Response` for example.
If you `dispatch` a `Response`, you don't care if it's a `JsonResponse`, a `BinaryFileResponse` or a `RedirectResponse`. These three classes haven't the same capabilities but are called with the same method =)

Reply:
As a note, we can't use dispatch() as it will conflict with EventDispatcherInterface.

For now, we want to keep the compatibility with the Event Dispatcher component, so we MUST NOT override this function and his signature.